src/Tokenizer/Transformer/BraceTransformer.php

```diff
@@ -195,6 +195,12 @@ private function transformIntoPropertyHookBraces(Tokens $tokens, int $index): vo
 
     private function transformIntoCurlyIndexBraces(Tokens $tokens, int $index): void
     {
+        // as $arr{index} is illegal since 8.4 - ParseError
+        // (and produces FatalError on 8.0-8.3)
+        if (\PHP_VERSION_ID >= 8_04_00) {
```

Comment:
Please help me understand why we skip transforming only for 8.4+ while this syntax does not work on 8.0+?

Reply:
https://3v4l.org/69BjL

please help to make comment describe it better

Reply:
I get that it behaves differently but I asked about something different: why we allow transformer on PHP 8.0-8.3 to replace tokens to custom tokens that represent the [concept that is invalid](https://3v4l.org/5Bt6U#v8.0.0)? It does not make sense to me. `CT::T_ARRAY_INDEX_CURLY_BRACE_*` should be available in the tokens collection only for PHP <8.0.

Reply:
It's a clear bug to me and I believe we rather need to fix tests if something fails when the condition is based on PHP 8.0.

Reply:
I believe this one is good as it reflects validation logic of built-in tokenizer.

if you want to extend it further to match `php -l` validation, I'm OKey with that. feel free to raise it as separated PR